activemodel/lib/active_model/naming.rb

```diff
@@ -195,28 +195,39 @@ def initialize(klass, namespace = nil, name = nil, locale = :en)
     #
     # Specify +options+ with additional translating options.
     def human(options = {})
-      return @human unless @klass.respond_to?(:lookup_ancestors) &&
-                           @klass.respond_to?(:i18n_scope)
-
-      defaults = @klass.lookup_ancestors.map do |klass|
-        klass.model_name.i18n_key
-      end
+      return @human if i18n_keys.empty? || i18n_scope.empty?
 
+      key, *defaults = i18n_keys
       defaults << options[:default] if options[:default]
-      defaults << @human
+      defaults << MISSING_TRANSLATION
```

Comment:
Why do we need this change?
This `MISSING_TRANSLATION` does not look like to contribute to the performance improvement.

Reply:
This change gives a large performance improvement.  Without this change the total improvement is much smaller:

```
Calculating -------------------------------------
              warmup   964.106k memsize (    23.647k retained)
                         2.156k objects (   311.000  retained)
                        50.000  strings (    50.000  retained)
               human     3.568k memsize (     0.000  retained)
                        27.000  objects (     0.000  retained)
                         6.000  strings (     0.000  retained)

Warming up --------------------------------------
               human   954.000  i/100ms
Calculating -------------------------------------
               human      9.273k (± 1.0%) i/s -     46.746k in   5.041362s
```

I agree, it is surprising!

I was planning to investigate the I18n code when I have time.  But I do not know whether improving the I18n code will be easy (for example, due to backwards compatibility).